docs/getting-started.md

```diff
@@ -2,4 +2,279 @@
+```
+
+This may even solve compilation issues you face when working with Substrate.
+
+## Front-End Developer Environment
+
+Substrate uses Yarn for all of it's front-end development needs. Visit the [install instructions](https://yarnpkg.com/en/docs/install) for Yarn to set up your computer. 
+
+Make sure you have the latest stable version of Node.js (`>= 10.16.3`) and Yarn (`>= 1.19.0`).
+
+## Get the Source
+
+Depending on what you are looking to build, there are some different starting points for Substrate development.
+
+> **Note:** You can add the `--help` flag to any command to see additional options.
+
+### Substrate
+
+The main Substrate project contains all core libraries which power Substrate-based chains, a pre-built node for running the Substrate testnet, and a key generation utility called Subkey.
+
+This starting point makes sense if you plan to contribute to the Substrate project or if you want to run the Substrate testnet or Subkey.
+
+Get the project at:
+
+```
+https://github.com/paritytech/substrate
+```
+
+Build the project with:
+
+```bash
+cargo build --release
+```
+
+Test the project with:
+
+```
+cargo test --all
+```
+
+Start the Substrate node with:
+
+```bash
+./target/release/substrate
+```
+
+#### Install the Substrate Node Locally
+
+You can install the Substrate node binary locally for easy access to running a node.
+
+In the Substrate project folder, run:
+
+```
+cargo install --force --path ./ substrate
+```
+
+You can then run this generated binary with:
+
+```bash
+substrate
+```
+
+#### Install Subkey Locally
+
+You can install the Subkey binary locally for easy access to this utility.
+
+In the Substrate project folder, run:
+
+```bash
+cargo install --force --path ./subkey subkey
+```
+
+You can then run this generated binary with:
+
+```bash
+subkey
+```
+
+### Substrate Node Template
+
+We provide a minimal working Substrate node meant for the development of new Substrate blockchains.
+
+Get the project at:
+
+```
+https://github.com/substrate-developer-hub/substrate-node-template
+```
+
+Build the project with:
+
+```bash
+cargo build --release
+```
+
+Purge any existing developer node state with:
+
+```bash
+# You will be prompted to type `y` to delete the database
+./target/release/node-template purge-chain --dev
+```
+
+Start a developer node with:
+
+```bash
+./target/release/node-template --dev
+```
+
+### Substrate Front-End Template
+
+We provide a minimal working Substrate front-end, built with ReactJS and the Polkadot-JS API. This project is meant for quick and easy development of custom user interfaces.
+
+Get the project at:
+
+```
+https://github.com/substrate-developer-hub/substrate-front-end-template
+```
+
+Install the node dependencies with:
+
+```bash
+yarn install
+```
+
+Run the front end with:
+
+```bash
+yarn start
+```
+
+Connect to the front-end at [`localhost:8000`](http://localhost:8000).
+
+> **Note:** You need to have a local Substrate node running to interact with this UI.
```

Comment:
Would be nice to link here to direction for getting a local Substrate node running.